core/gather/driver/execution-context.js

```diff
@@ -82,15 +89,25 @@ class ExecutionContext {
       this._session.getNextProtocolTimeout() :
       60000;
 
+    // `__lighthouseExecutionContextUniqueIdentifier` is only used by the FullPageScreenshot gatherer.
+    // See `getNodeDetails` in page-functions.
+    const uniqueExecutionContextIdentifier = contextId === undefined ?
+      0 :
+      this._executionContextIdentifiersCreated;
+
```

Comment:
@brendankenny not sure if this is what you had in mind, but we need to be mindful of the "undefined"/"page" context here.

Reply:
Yeah, more or less, though I actually think the previous `undefined` (instead of 0) here is kind of cleaner--there is no isolated context for that case, it makes it clearer that it's separate from the numbering of the contexts, and it eliminates the need for the `'?'` case below.